lib/Raven/Client.php

```diff
@@ -231,76 +110,67 @@ class Client
     /**
      * @var bool
      */
-    protected $_shutdown_function_has_been_set;
+    protected $_shutdown_function_has_been_set = false;
+
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
Again, the constructor of the client will necessarily need to accept new arguments in the near future, when factories will be introduced, so it will not stay so simple anyway. There will be no way the constructor will stay as is